*** ./CMakeLists.txt	2013-09-16 20:47:51.000000000 +0200
--- ./CMakeLists.txt	2014-04-23 17:47:59.403087345 +0200
***************
*** 291,296 ****
--- 291,297 ----
  # Passing just the major version works with as desired multi-python-capable find module such
  # as in latest CMake (2.8.9)
  find_package(PythonLibs ${_VRPN_PYTHON_VERSIONSEARCH})
+ set(PYTHONLIBS_FOUND OFF)
  if(PYTHONLIBS_FOUND)
  	if(PYTHONLIBS_VERSION_STRING)
  		string(SUBSTRING
