*** ./quat/CMakeLists.txt	2014-04-23 18:16:52.039124174 +0200
--- ./quat/CMakeLists.txt	2014-04-23 18:17:21.103124792 +0200
***************
*** 35,40 ****
--- 35,41 ----
 add_library(quat ${QUATLIB_SOURCES} ${QUATLIB_HEADER})
 set_property(TARGET quat PROPERTY PUBLIC_HEADER ${QUATLIB_HEADER})
 set_target_properties(quat PROPERTIES FOLDER Library)
+set_target_properties(quat PROPERTIES VERSION 7.33)
 if(VRPN_INSTALL)
 	install(TARGETS
  		quat
*** ./CMakeLists.txt	2014-04-23 18:18:26.147126174 +0200
--- ./CMakeLists.txt	2014-04-23 18:21:56.727130650 +0200
***************
*** 1162,1167 ****
--- 1162,1168 ----
  
  if(VRPN_BUILD_CLIENT_LIBRARY)
  	add_library(vrpn ${VRPN_CLIENT_SOURCES} ${VRPN_CLIENT_PUBLIC_HEADERS})
+ 	set_target_properties(vrpn PROPERTIES VERSION 7.33)
  	target_link_libraries(vrpn ${EXTRA_LIBS})
  	set(VRPN_CLIENT_LIBRARY vrpn)

