*** lib/bluetooth.h.old	2015-06-01 16:09:38.803019017 +0200
--- lib/bluetooth.h	2015-06-01 16:10:04.839017928 +0200
***************
*** 140,155 ****
  #define bt_get_unaligned(ptr)			\
  ({						\
  	struct __attribute__((packed)) {	\
! 		typeof(*(ptr)) __v;		\
! 	} *__p = (typeof(__p)) (ptr);		\
  	__p->__v;				\
  })
  
  #define bt_put_unaligned(val, ptr)		\
  do {						\
  	struct __attribute__((packed)) {	\
! 		typeof(*(ptr)) __v;		\
! 	} *__p = (typeof(__p)) (ptr);		\
  	__p->__v = (val);			\
  } while(0)
  
--- 140,155 ----
  #define bt_get_unaligned(ptr)			\
  ({						\
  	struct __attribute__((packed)) {	\
! 		__typeof__(*(ptr)) __v;		\
! 	} *__p = (__typeof__(__p)) (ptr);		\
  	__p->__v;				\
  })
  
  #define bt_put_unaligned(val, ptr)		\
  do {						\
  	struct __attribute__((packed)) {	\
! 		__typeof__(*(ptr)) __v;		\
! 	} *__p = (__typeof__(__p)) (ptr);		\
  	__p->__v = (val);			\
  } while(0)
  
