--- opencv-1.1.0/cxcore/src/cxmathfuncs.cpp	2017-04-07 17:02:43.541371640 +0200
+++ opencv-1.1.0/cxcore/src/cxmathfuncs.cpp	2017-04-07 16:49:56.568211660 +0200
@@ -62,7 +62,7 @@
 };
 
 static const int icvAtanSign[8] =
-    { 0, 0x80000000, 0x80000000, 0, 0x80000000, 0, 0, 0x80000000 };
+    { 0, (int)0x80000000, (int)0x80000000, 0, (int)0x80000000, 0, 0, (int)0x80000000 };
 
 CV_IMPL float
 cvFastArctan( float y, float x )
